### PR TITLE
Changes from livity deployment

### PR DESCRIPTION
**Pull request by rjmackay**
This pull request makes the following changes:
- Decode twitter user's profile location
- add bounding box of user's place too
- Allow saving post with attributes that have no form
- Twitter: Skip retweets when pulling messages (#427)

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform/1234)

### CODE REVIEW

File: application/classes/Ushahidi/Repository/Form/Attribute.php [hunk 1]
```diff
@@ -73,16 +73,23 @@ public function getSearchFields()
 	}
 
 	// FormAttributeRepository
-	public function getByKey($key, $form_id = null)
+	public function getByKey($key, $form_id = null, $include_no_form = false)
```
Comment (rjmackay) [inline application/classes/Ushahidi/Repository/Form/Attribute.php]: Not 100% that I like the added param here. But no better ideas..

File: plugins/twitter/classes/DataProvider/Twitter.php [hunk 2]
```diff
@@ -104,6 +108,11 @@ public function fetch($limit = FALSE) {
 
 				$additional_data = [];
 
+				// Skip retweets
+				if ($status['retweeted_status']) {
```
Comment (rjmackay) [inline plugins/twitter/classes/DataProvider/Twitter.php]: This was easier than expected :)

File: plugins/twitter/classes/DataProvider/Twitter.php [hunk 3]
```diff
@@ -109,7 +109,7 @@ public function fetch($limit = FALSE) {
 				$additional_data = [];
 
 				// Skip retweets
-				if ($status['retweeted_status']) {
+				if (array_key_exists('retweeted_status', $status) && array_key_exists('text', $status['retweeted_status'])) {
```
Comment (rjmackay) [inline plugins/twitter/classes/DataProvider/Twitter.php]: @tuxpiper is this basically `!empty($status['retweeted_status']`? or is it specifically the `text` key that matters?

File: application/classes/Ushahidi/Validator/Post/Create.php [hunk 4]
```diff
@@ -183,7 +183,10 @@ public function checkValues(Validation $validation, $attributes, $data)
 		foreach ($attributes as $key => $values)
 		{
 			// Check attribute exists
-			$attribute = $this->attribute_repo->getByKey($key, $data['form_id']);
+			$attribute = $this->attribute_repo->getByKey($key, $data['form_id'], true);
+			if (! $attribute->id) {
+
```
Comment (willdoran) [inline application/classes/Ushahidi/Validator/Post/Create.php]: Is this repetition intentional?

**Pull request merged**